File: my-YBO-app/server/routes/authRoutes.js
// my-YBO-app/server/routes/authRoutes.js
const express = require("express");
const bcrypt = require("bcrypt");
const db = require("../db/db");
const router = express.Router();

router.post("/signup", async (req, res) => {
  try {
    const { email, password, name } = req.body;
    const normalizedEmail = email.trim().toLowerCase();
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

    if (!emailRegex.test(normalizedEmail)) {
      return res.status(400).json({
        error: "Invalid email format",
      });
    }

    const [existingUsers] = await db.query(
      "SELECT id FROM users WHERE email = ?",
      [normalizedEmail]
    );

    if (existingUsers.length > 0) {
      return res.status(400).json({
        error: "Email already exists",
      });
    }

    const hashedPassword = await bcrypt.hash(password, 10);
    const displayName = name || normalizedEmail.split("@")[0];
    const profilePicture = `https://api.dicebear.com/9.x/adventurer/svg?seed=${encodeURIComponent(
      displayName
    )}`;

    const sql = `
      INSERT INTO users 
      (email, password, name, bio, profile_picture) 
      VALUES (?, ?, ?, ?, ?)
    `;

    const [result] = await db.query(sql, [
      normalizedEmail,
      hashedPassword,
      displayName,
      "New user",
      profilePicture,
    ]);

    res.status(201).json({
      message: "User created",
      user: {
        id: result.insertId,
        email: normalizedEmail,
        name: displayName,
        bio: "New user",
        profile_picture: profilePicture,
      },
    });
  } catch (err) {
    console.error("Signup failed:", err);
    res.status(500).json({
      error: err.message,
    });
  }
});

router.post("/login", async (req, res) => {
  try {
    const { email, password } = req.body;
    const normalizedEmail = email.trim().toLowerCase();
    const [users] = await db.query(
      "SELECT * FROM users WHERE email = ?",
      [normalizedEmail]
    );

    
    if (users.length === 0) {
      return res.status(401).json({
        error: "Invalid credentials",
      });
    }

    const user = users[0];
    const isMatch = await bcrypt.compare(password, user.password);

    if (!isMatch) {
      return res.status(401).json({
        error: "Invalid credentials",
      });
    }

    res.json({
      message: "Login successful",
      user: {
        id: user.id,
        email: user.email,
        name: user.name,
        bio: user.bio,
        profile_picture: user.profile_picture,
      },
    });
  } catch (err) {
    console.error("Login failed:", err);
    res.status(500).json({
      error: err.message,
    });
  }
});

module.exports = router;

File: my-YBO-app/backend/app/routes/auth_routes.py
from flask import Blueprint, request, jsonify
from app.services import AuthService

auth_bp = Blueprint('auth', __name__, url_prefix='/api/auth')


@auth_bp.route('/signup', methods=['POST'])
def signup():
    """Register a new user"""
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({'error': 'No data provided'}), 400
        
        email = data.get('email')
        password = data.get('password')
        name = data.get('name')
        
        if not email or not password:
            return jsonify({'error': 'Email and password are required'}), 400
        
        result = AuthService.signup(email, password, name)
        
        if result['success']:
            return jsonify({
                'message': 'User created',
                'user': result['user']
            }), 201
        else:
            return jsonify({'error': result['error']}), 400
    
    except Exception as e:
        return jsonify({'error': str(e)}), 500


@auth_bp.route('/login', methods=['POST'])
def login():
    """Authenticate a user"""
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({'error': 'No data provided'}), 400
        
        email = data.get('email')
        password = data.get('password')
        
        if not email or not password:
            return jsonify({'error': 'Email and password are required'}), 400
        
        result = AuthService.login(email, password)
        
        if result['success']:
            return jsonify({
                'message': 'Login successful',
                'user': result['user']
            }), 200
        else:
            return jsonify({'error': result['error']}), 401
    
    except Exception as e:
        return jsonify({'error': str(e)}), 500
